src/Raven.Embedded/RavenEmbedded.cs

```diff
@@ -0,0 +1,207 @@
+                await store.Maintenance.Server.SendAsync(new CreateDatabaseOperation(new DatabaseRecord(database))).ConfigureAwait(false);
+            }
+            catch (ConcurrencyException)
+            {
+                // Expected behaviour when the database is already exists
+
+                if (Logger.IsInfoEnabled)
+                    Logger.Info($"{database} already exist.");
+            }
+
+            _documentStores.TryAdd(database, store);
+            return store;
+
+        }
+
+        public async Task<Uri> GetServerUri()
+        {
+            if (Interlocked.Read(ref _initialized) == 0)
+                throw new InvalidOperationException("Please run Setup before trying to get the server uri");
+
+            return await _serverTask.ConfigureAwait(false);
+        }
+
+        public void RemoveStore(string database)
+        {
+            if(_documentStores.TryRemove(database, out var store))
+                store.Dispose();
+        }
+
+        public void KillSlavedServerProcess()
+        {
+            var p = _ravenDbSlavedProcess;
+            _ravenDbSlavedProcess = null;
+
+            if (p == null || p.HasExited != false)
+                return;
+
+            if (Logger.IsInfoEnabled)
+                Logger.Info($"Kill global server PID { p.Id }.");
+
+            try
+            {
+                p.Kill();
+                Interlocked.CompareExchange(ref _initialized, 0, 1);
+            }
+            catch (Exception e)
+            {
+                if (Logger.IsInfoEnabled)
+                {
+                    Logger.Info($"{DateTime.Now}: {e}\r\n");
+                }
+
+                if (p.HasExited)
+                    Interlocked.CompareExchange(ref _initialized, 0, 1);
+
+                throw;
+            }
+            finally
+            {
+                foreach (var store in _documentStores.Values)
+                {
+                    store.Dispose();
+                }
+                _documentStores.Clear();
+            }
+        }
+
+        private async Task<Uri> RunServer(ServerOptions options)
+        {
+            var process = _ravenDbSlavedProcess = RavenServerRunner.Run(options);
+            if (Logger.IsInfoEnabled)
+            {
+                Logger.Info($"Starting global server: { _ravenDbSlavedProcess.Id }");
+            }
+
+            string url = null;
+            var output = process.StandardOutput;
+            var sb = new StringBuilder();// listen to standard error as well
+
+            var startupDuration = Stopwatch.StartNew();
+
+            Task<string> readLineTask = null;
+            while (true)
+            {
+                if (readLineTask == null)
+                    readLineTask = output.ReadLineAsync();
+
+                var hasResult = await readLineTask.WaitWithTimeout(TimeSpan.FromSeconds(5)).ConfigureAwait(false);
+
+                if (startupDuration.Elapsed > TimeSpan.FromMinutes(1)) //from options
```

Comment:
I assume that the comment means TODO? But it is missing that and it is quite misleading